documentation/dataplane/dash-flow-api.md

```diff
@@ -0,0 +1,633 @@
+| SAI_FLOW_TABLE_ATTR_DASH_FLOW_ENABLED_KEY | `sai_dash_flow_enabled_key_t` | Key enable mask |
+| SAI_FLOW_TABLE_ATTR_FLOW_TTL_IN_MILLISECONDS | `sai_uint32_t`     | Time-to-live (TTL) for flows in milliseconds.  |
+
+The sai_dash_flow_enabled_key_t is defined as follows:
+
+```c
+typedef enum _sai_dash_flow_enabled_key_t
+{
+    SAI_DASH_FLOW_ENABLED_KEY_NONE = 0,
+
+    SAI_DASH_FLOW_ENABLED_KEY_ENI_ADDR = 1 << 1,
+
+    SAI_DASH_FLOW_ENABLED_KEY_PROTOCOL = 1 << 2,
+
+    SAI_DASH_FLOW_ENABLED_KEY_SRC_IP = 1 << 3,
+
+    SAI_DASH_FLOW_ENABLED_KEY_DST_IP = 1 << 4,
+
+    SAI_DASH_FLOW_ENABLED_KEY_SRC_PORT = 1 << 5,
+
+    SAI_DASH_FLOW_ENABLED_KEY_DST_PORT = 1 << 6,
+
+} sai_dash_flow_enabled_key_t;
+```
+
+## Flow APIs
+
+### Basic flow APIs
+
+The flow_entry APIs are defined as follows:
+
+| API                      | Description                                                  |
+| ------------------------ | :----------------------------------------------------------- |
+| create_flow_entry        | Add a single new entry to a certain flow table               |
+| remove_flow_entry        | Remove a single entry in a certain flow table. Note that the flow removal process deletes two flows if it is a bi-directional flow. If you wish to remove a flow in only one direction, you should set the flow to be uni-directional in advance. |
+| set_flow_entry_attribute | Set attributes for a single entry in a certain flow table    |
+| get_flow_entry_attribute | Get attributes of a single entry in a certain flow table     |
+| create_flow_entries      | Add multiple entries to a certain flow table in bulk         |
+| remove_flow_entries      | Remove multiple entries from a specific flow table in bulk. Note that the flow removal process deletes two flows if it is a bi-directional flow. If you wish to remove a flow in only one direction, you should set the flow to be uni-directional in advance. |
+
+### Keys of flow entry
+
+The keys for a flow entry are defined as follows:
+
+Please note that there is an attribute in the *flow_table* that can specify which of the following keys are enabled. If a key is not enabled, it will not be used in match and action.
+
+The *flow_table_id* is used to designate the flow table for the flow only, which is not used in match and action.
+
+```c
+typedef struct _sai_flow_entry_t
+{
+    /**
+     * @brief Switch ID
+     *
+     * @objects SAI_OBJECT_TYPE_SWITCH
+     */
+    sai_object_id_t switch_id;
+
+    /**
+     * @brief Exact matched key flow_table_id
+     *
+     * @objects SAI_OBJECT_TYPE_FLOW_TABLE
+     */
+    sai_object_id_t flow_table_id;
+
+    /**
+     * @brief Exact matched key eni_mac
+     */
+    sai_mac_t eni_mac;
+
+    /**
+     * @brief Exact matched key ip_protocol
+     */
+    sai_uint8_t ip_proto;
+
+    /**
+     * @brief Exact matched key src_ip_addr
+     */
+    sai_ip_address_t src_addr;
```

Comment:
i believe this is missed. since we changed the reverse flow key to src ip and dst ip, here will be better to do the same thing.